monai/data/dataset.py

```diff
@@ -850,21 +850,20 @@ def _load_cache_item(self, idx: int):
         return item
 
     def _transform(self, index: int):
-        index_: Any = index
         if self.hash_as_key:
             key = self.hash_func(self.data[index])
-            if key in self._cache:
+            if key in self._hash_keys:
                 # if existing in cache, get the index
-                index_ = key  # if using hash as cache keys, set the key
+                index = self._hash_keys.index(key)
 
-        if isinstance(index_, int) and index_ % len(self) >= self.cache_num:  # support negative index
+        if index % len(self) >= self.cache_num:  # support negative index
             # no cache for this index, execute all the transforms directly
-            return super()._transform(index_)
+            return super()._transform(index)
         # load data from cache and execute from the first random transform
         start_run = False
         if self._cache is None:
```

Comment:
this line is not part of the PR, but out of curiosity, do we expect _cache to be None at this point?   I thought we always set it in init / set_data at the beginning.

Reply:
You are right, you can see I already did the change in https://github.com/Project-MONAI/MONAI/pull/5365, so I don't want to make duplicated change in this PR.
Anyway, I already updated it in the latest commit.

Thanks.